### ISSUE TITLE
torch.tril and torch.triu produce incorrect results with device='cuda'

### ISSUE THREAD

**Issue by mrshenli** [pytorch/pytorch#15226]
## 🐛 Bug

The results of `torch.tril` and `torch.triu` are reversed on `device='cuda'`. (correct on 'cpu')

## To Reproduce

```
>>> torch.ones(2028, 1, dtype=torch.long, device='cuda').triu(2)
tensor([[0],
        [0],
        [1],
        ...,
        [1],
        [1],
        [1]], device='cuda:0')
>>> torch.ones(2028, 1, dtype=torch.long, device='cpu').triu(2)
tensor([[0],
        [0],
        [0],
        ...,
        [0],
        [0],
        [0]])
>>> torch.ones(2028, 1, dtype=torch.long, device='cuda').tril(2)
tensor([[1],
        [1],
        [1],
        ...,
        [0],
        [0],
        [0]], device='cuda:0')
>>> torch.ones(2028, 1, dtype=torch.long, device='cpu').tril(2)
tensor([[1],
        [1],
        [1],
        ...,
        [1],
        [1],
        [1]])
```

## Expected behavior

```
>>> torch.ones(2028, 1, dtype=torch.long, device='cuda').triu(2)
tensor([[1],
        [1],
        [1],
        ...,
        [0],
        [0],
        [0]], device='cuda:0')
>>> torch.ones(2028, 1, dtype=torch.long, device='cuda').tril(2)
tensor([[0],
        [0],
        [1],
        ...,
        [1],
        [1],
        [1]], device='cuda:0')
```

## Environment
PyTorch version: 1.0.0a0+62178ba
Is debug build: No
CUDA used to build PyTorch: 9.2.88

OS: CentOS Linux 7 (Core)
GCC version: (GCC) 4.8.5 20150623 (Red Hat 4.8.5-28)
CMake version: version 3.12.2

Python version: 3.6
Is CUDA available: Yes
CUDA runtime version: 9.2.88
GPU models and configuration: 
GPU 0: Tesla M40
GPU 1: Tesla M40

Nvidia driver version: 396.26
cuDNN version: Probably one of the following:
/usr/local/cuda-8.0/targets/x86_64-linux/lib/libcudnn.so.5.0.5
/usr/local/cuda-8.0/targets/x86_64-linux/lib/libcudnn.so.5.1.3
/usr/local/cuda-8.0/targets/x86_64-linux/lib/libcudnn.so.6.0.20
/usr/local/cuda-8.0/targets/x86_64-linux/lib/libcudnn.so.6.0.21
/usr/local/cuda-8.0/targets/x86_64-linux/lib/libcudnn_static.a
/usr/local/cuda-9.2/targets/x86_64-linux/lib/libcudnn.so.7.1.2
/usr/local/cuda-9.2/targets/x86_64-linux/lib/libcudnn.so.7.4.1
/usr/local/cuda-9.2/targets/x86_64-linux/lib/libcudnn_static.a
/usr/local/fbcode/gcc-5-glibc-2.23/lib/libcudnn.so.6
/usr/local/fbcode/gcc-5-glibc-2.23/lib/libcudnn.so.6.0.20
/usr/local/fbcode/gcc-5-glibc-2.23/lib/libcudnn.so.6.0.21
/usr/local/fbcode/gcc-5-glibc-2.23/lib/libcudnn.so.7.1.2
/usr/local/fbcode/gcc-5-glibc-2.23/lib/libcudnn.so.7.1.4
/usr/local/fbcode/platform007/lib/libcudnn.so.7.1.4

Versions of relevant libraries:
[pip] Could not collect
[conda] blas                      1.0                         mkl  
[conda] magma-cuda92              2.4.0                         1    pytorch
[conda] mkl                       2018.0.3                      1  
[conda] mkl-include               2019.1                      144  
[conda] mkl_fft                   1.0.1            py36h3010b51_0  
[conda] mkl_random                1.0.1            py36h629b387_0  
[conda] torch                     1.0.0a0+62178ba           <pip>

**Comment by zou3519** [comment]
cc @vishwakftw do your batched triu/tril implementations fix this?

**Comment by mrshenli** [comment]
fixed by #15257
